airbyte/cli.py

```diff
@@ -0,0 +1,392 @@
+
+    def _inject_secrets(config_dict: dict[str, Any]) -> None:
+        """Inject secrets into the configuration dictionary."""
+        for key, value in config_dict.items():
+            if isinstance(value, dict):
+                _inject_secrets(value)
+            elif isinstance(value, str) and value.startswith("SECRET:"):
+                config_dict[key] = get_secret(value.removeprefix("SECRET:").strip())
+
+    config_dict: dict[str, Any]
+    if config.startswith("{"):
+        # Treat this as an inline yaml string:
+        config_dict = yaml.safe_load(config)
+    else:
+        # Treat this as a path to a config file:
+        config_path = Path(config)
+        if not config_path.exists():
+            raise PyAirbyteInputError(
+                message="Config file not found.",
+                input_value=str(config_path),
+            )
+        config_dict = json.loads(config_path.read_text(encoding="utf-8"))
+
+    _inject_secrets(config_dict)
+    return config_dict
+
+
+def _resolve_source_job(
+    *,
+    source: str | None = None,
+    config: str | None = None,
+    streams: str | None = None,
+) -> Source:
+    """Resolve the source job into a configured Source object.
+
+    Args:
+        source: The source name, with an optional version declaration.
+            If a path is provided, the source will be loaded from the local path.
+            If the string `'.'` is provided, the source will be loaded from the current
+            working directory.
+        config: The path to a configuration file for the named source or destination.
+        streams: A comma-separated list of stream names to select for reading. If set to "*",
+            all streams will be selected. If not provided, all streams will be selected.
+    """
+    source_obj: Source
+    if source and (source.startswith(".") or "/" in source):
+        # Treat the source as a path.
+        source_executable = Path(source)
+        if not source_executable.exists():
+            raise PyAirbyteInputError(
+                message=f"Source executable not found: {source}",
+            )
+        source_obj = get_source(
+            name=source_executable.stem,
+            local_executable=source_executable,
+        )
+        return source_obj
+    if not config:
+        raise PyAirbyteInputError(
+            message="No configuration found.",
+        )
+    if not source or not source.startswith("source-"):
+        raise PyAirbyteInputError(
+            message="Expected a source name or path to executable.",
+            input_value=source,
+        )
+
+    source_name: str = source
+    streams_list: str | list[str] = streams or "*"
+    if isinstance(streams, str) and streams != "*":
+        streams_list = [stream.strip() for stream in streams.split(",")]
+
+    return get_source(
+        name=source_name,
+        config=_resolve_config(config) if config else {},
+        streams=streams_list,
+    )
```

Comment:
@coderabbitai - The guide for PyAirbyte error messaging is inspired by structlog, decoupling the message and the dynamic values affecting the scenario. In this case, we already provide the input value to the exception constructor, which accomplishes the needed messaging to the user with `input_value=source` - and we don't need to put the value in the message text.

This also enables us to be more sensitive to PII when sending telemetry, since `message` will _never_ contain PII. We can log in telemetry the message string without the `input_value`, with no concern for PII leakage.